andela/shadowcat-backend
# #167729796 Implement Notification feature

#### What does this PR do?
Implementation of the Notification Feature

#### Description of Task to be completed?
Upon successful booking of a trip, a user should get an email notification and the user's manager should get an in-app notification of the trip

#### How should this be manually tested?
This feature is dependent on the create-trip route, so to test it follow these steps carefully

- Pull from the branch
- Navigate to **localhost:3000//api/v1/trips/get_trips/:id**  where **id** is the manager Id of the user that wants to create trip
- Navigate to the login route from your postman and Login
- Copy and keep the token
- Navigate to **http://localhost:3000/api/v1/trips/request** and pass in the trip details
- Ensure to pass in the **Bearer token** in the header of the request as **Authorization**
![Pr](https://user-images.githubusercontent.com/50335408/64695557-581c2a80-d494-11e9-896a-68f6da39bd9f.png)

- Ensure to have correct info in the details
![PR-prrof](https://user-images.githubusercontent.com/50335408/64695524-42a70080-d494-11e9-8fa3-139a0ae02776.png)

- You should receive a success message
- Check your email to see the trip confirmation
- Go back to your browser to confirm that the trip details are showing for the manager without refreshing

#### Any background context you want to provide?
Ensure that the create-trip feature has been merged before testing

#### What are the relevant pivotal tracker stories?
#167729796

#### Screenshots (if appropriate)
![proof_notify](https://user-images.githubusercontent.com/50335408/64529266-6a665f00-d302-11e9-9ebd-cf0274cd8bfe.png)
![proof_notify2](https://user-images.githubusercontent.com/50335408/64529321-85d16a00-d302-11e9-9ec5-d500fb795e20.png)


#### Questions:
None
